DOCSP-42300: Standardize Monitoring page (#82)
mongodb/docs-java-rs: source/monitoring.txt

   The descriptions of JMX and JConsole in this example are illustrative
   rather than a source of truth. For guaranteed up to date information, consult
   the following official Oracle resources:

   - `JConsole documentation <https://www.oracle.com/technical-resources/articles/java/jconsole.html>`__
   - `JMX documentation <https://docs.oracle.com/javase/tutorial/jmx/index.html>`__

The following code snippet adds a ``JMXConnectionPoolListener`` to a
``MongoClient`` instance. The code then pauses execution so you can
navigate to JConsole and inspect your connection pools.

.. io-code-block::
   :copyable: true

   .. input:: /includes/monitoring/JMXMonitoring.java
      :language: java
      :start-after: start-jmx-example
      :end-before: end-jmx-example
      :dedent:

   .. output::
      :visible: false

      Navigate to JConsole to see your connection pools...

Once you have started your server, open JConsole by running the following command in your
terminal:

.. code-block:: shell

   jconsole

Once JConsole is open, perform the following actions in the GUI:

#. Select the Java process running the preceding example code.

> #. Press :guilabel:`Insecure Connection` in the warning dialog box.

Reviewer: [s] In the style guide, it sounds like `press` is reserved for a specific key on a keyboard, but not super familiar with JConsole so possibly that is a better choice so might not need to update:
```suggestion
#. Click :guilabel:`Insecure Connection` in the warning dialog box.
```

#. Click on the :guilabel:`MBeans` tab.
#. Inspect your connection pool events under the ``"org.mongodb.driver"`` domain.

> When you no longer want to inspect your connection pools in JConsole, do the
> following:

Reviewer: [s] To better align with the introducing lists section of the style guide: 
```suggestion
When you no longer want to inspect your connection pools in JConsole, perform the
following tasks:
```

> - Exit JConsole by closing the JConsole window
> - Stop the Java program running the preceding code snippet

Reviewer: [q] Do these need to be completed in order or it doesn't matter? Also I think can use punctuation as imperative statements.
```suggestion
- Exit JConsole by closing the JConsole window.
- Stop the Java program running the preceding code snippet.
```
